lib/pbench/test/unit/server/query_apis/test_datasets_detail.py

```diff
@@ -341,11 +269,8 @@ def test_nonunique_query(
         """
         Check the handling of a query that returns too much data.
         """
-        json = {
-            "name": "fio",
-            "start": "2020-08",
-            "end": "2020-10",
-        }
+        # We remove the user value so we can make an unauthorized query
+        del self.payload["user"]
```

Comment:
Is this current instance of this class, upon which this member function is invoked, reinstantiated for each test scenario in it?  (I'm concerned that removing the `"user"` field from the `payload` will have an impact on subsequent invocations of this scenario or others in the test-class...and, I'm wondering if this code should restore the field when it is done....)

Reply:
I thought about suggesting the copy approach, but I thought I'd first try to find out whether we even need it.  (Copying objects in Python is notoriously error-prone -- even `copy()` isn't necessarily sufficient to protect the original, depending on what needs to be done to the copy -- it might need `deepcopy()`, but, of course, even _that_ doesn't necessarily do what one wants.... 😝)

Reply:
Good point but the fixture scope of the `_setup` is "function" (its default value) and so the fixture is destroyed at the end of the each test function. And since the fixture is defined with `autouse=true` it will be available at the beginning of each test as it was defined in `_setup`.

Reply:
I'm less worried about the _fixture_ than I am about the test class instance; but, if the fixture completely reinitializes the instance on each test-case (function) invocation, then I guess we're good.

Reply:
Ah; it seems inefficient and generally unnecessary to build a new instance for every test method, but it does eliminate the risk of cross-contamination and it's not like our tests are taking "hours" to complete. So, carry on ... 😆